### ISSUE TITLE
Bundler/DuplicatedGem cop doesn't understand if/else blocks

### ISSUE THREAD

**Issue by wollkind** [bbatsov/rubocop#4432]
We have a section of our gemfile that looks roughly like this:
```
 if some_condition
    gem 'some-gem', '~> 5'
  else
    gem 'some-gem', '~> 4'
  end
```

Rubocop erroneously flags this as a duplicated gem

**Comment by bbatsov** [comment]
Supporting this would complicate the code a lot and I'm not certain that's worth it or not. I've been programming in Ruby for over 10 years and I've never had to use conditional deps, so it seems to me that's not very common.

**Comment by backus** [comment]
This is a dupe of https://github.com/bbatsov/rubocop/issues/3752 where we have had some of this conversation already
